app/models/token_auth/synchronizable_resource.rb

```diff
@@ -0,0 +1,14 @@
+module TokenAuth
+  # A resource that may be pushed and/or pulled by an entity.
+  class SynchronizableResource < ActiveRecord::Base
+    include UuidEnabled
+
+    validates :uuid, :entity_id, :entity_id_attribute_name, :name, :class_name,
+              presence: true
+    validates :is_pullable, :is_pushable, inclusion: { in: [true, false] }
+
+    def klass
+      class_name.constantize
```

Comment:
I probably need a validation in here for `class_name` anyway to make sure the referenced class exists.